ibis/pyspark/compiler.py

```diff
@@ -203,6 +235,13 @@ def compile_literal(t, expr, scope, raw=False, **kwargs):
         return F.lit(expr.op().value)
 
 
+def _compile_agg(t, scope, agg_expr, context):
+    agg = t.translate(agg_expr, scope, context=context)
+    if agg_expr.has_name():
```

Comment:
When does `has_name` return True/False?

Reply:
Should return true when the expression has a column name and false otherwise. When there's no name, I've seen `tmp` in the rendering of the expression.

Reply:
Ok.. `tmp` should not be a user facing thing. I'd prefer we don't return `tmp` as column names to user unless it's an ibis convention.

Reply:
I think that's part of the expected result in some cases though. E.g. https://github.com/ibis-project/ibis/blob/master/ibis/tests/all/test_aggregation.py#L145

Reply:
well, in the case that @hjoo mentions, it's named `'tmp`' because of the call to `aggregate` above. we should try to find where it will attempt to name a column tmp in the actual code.

Reply:
Default name for `reduction_to_aggregation()` is `tmp`. I think that would explain why the expected aggregation result is also named `tmp`.

Reply:
sounds good to me